### PR TITLE
GraphQL Schema for Gift Wrapping and Gift Message

### PR DESCRIPTION
**Pull request by paliarush**
GraphQL Schema for Gift Wrapping and Gift Message

### CODE REVIEW

File: design-documents/graph-ql/coverage/gift-wrapping.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+type CartPrices {
+    gift_options: GiftOptionsPrices
+}
+###### End: Extending existing types ######
+
+
+###### Begin: Defining new types ######
+type GiftWrapping {
+    id: ID!
```
Comment (joni-jones) [inline design-documents/graph-ql/coverage/gift-wrapping.md]: How id will be represented?

File: design-documents/graph-ql/coverage/gift-wrapping.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+}
+
+type GiftOptionsPrices {
+    gift_wrapping_for_order: Money
+    gift_wrapping_for_items: Money
+    printed_card: Money
+}
+
+type GiftMessage {
+    to: String!
```
Comment (joni-jones) [inline design-documents/graph-ql/coverage/gift-wrapping.md]: Could you add doc blocks? Will it be a recipient email or name?

File: design-documents/graph-ql/coverage/gift-wrapping.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    from: String!
+    message: String!
+}
+###### End: Defining new types ######
+
+```
+
+## Configuration
+
+The following gift options need to be whitelisted in the `storeConfig` query.
```
Comment (joni-jones) [inline design-documents/graph-ql/coverage/gift-wrapping.md]: Could you add a description what modules should be updated?

File: design-documents/graph-ql/coverage/gift-wrapping.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+```xml
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Config:etc/system_file.xsd">
```
Comment (joni-jones) [inline design-documents/graph-ql/coverage/gift-wrapping.md]: How does this configuration affect the flow if a merchant does not specify these options?

**Comment by joni-jones** [comment]
@paliarush could you add doc block to the graphql schema?

File: design-documents/graph-ql/coverage/gift-wrapping.md (hunk 1 above)
Comment (paliarush) [reply design-documents/graph-ql/coverage/gift-wrapping.md]: Can be base64 from the DB increment ID to be consistent with other entities.

File: design-documents/graph-ql/coverage/gift-wrapping.md (hunk 2 above)
Comment (paliarush) [reply design-documents/graph-ql/coverage/gift-wrapping.md]: Done.

File: design-documents/graph-ql/coverage/gift-wrapping.md (hunk 3 above)
Comment (paliarush) [reply design-documents/graph-ql/coverage/gift-wrapping.md]: Done.

File: design-documents/graph-ql/coverage/gift-wrapping.md (hunk 4 above)
Comment (paliarush) [reply design-documents/graph-ql/coverage/gift-wrapping.md]: These options are currently used in Luma, so the behavior should be the same. There are usually some defaults.

File: design-documents/graph-ql/coverage/gift-wrapping.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+## Data
+
+```graphql
+
+###### Begin: Extending existing types ######
+type Cart {
+    available_gift_wrappings: [GiftWrapping]! @doc(description: "The list of available gift wrapping options for the cart")
+    gift_wrapping: GiftWrapping @doc(description: "The selected gift wrapping for the cart")
+    include_printed_card: Boolean! @doc(description: "Wether customer requested printed card for the order")
```
Comment (paliarush) [inline design-documents/graph-ql/coverage/gift-wrapping.md]: Make declarative.

File: design-documents/graph-ql/coverage/gift-wrapping.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+}
+
+type GiftOptionsPrices {
+    gift_wrapping_for_order: Money @doc(description: "Price of the gift wrapping for the whole order")
+    gift_wrapping_for_items: Money @doc(description: "Price of the gift wrapping for all individual order items")
+    printed_card: Money @doc(description: "Price for the printed card")
+}
+
+type GiftMessage {
+    to: String! @doc(description: "Recepient name")
```
Comment (paliarush) [inline design-documents/graph-ql/coverage/gift-wrapping.md]: Change to `recipient` and `sender`

File: design-documents/graph-ql/coverage/gift-wrapping.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+}
+
+type SalesItemInterface {
+    gift_wrapping: GiftWrapping @doc(description: "The selected gift wrapping for the order item")
+    gift_message: GiftMessage @doc(description: "The entered gift message for the order item")
+}
+
+type CustomerOrder {
+    gift_wrapping: GiftWrapping @doc(description: "The selected gift wrapping for the order")
+    include_printed_card: Boolean! @doc(description: "Wether customer requested printed card for the order")
```
Comment (paliarush) [inline design-documents/graph-ql/coverage/gift-wrapping.md]: Wether => Whether

File: design-documents/graph-ql/coverage/gift-wrapping.md [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+    include_gift_receipt: Boolean! @doc(description: "Wether customer requested gift receipt for the order")
```
Comment (paliarush) [inline design-documents/graph-ql/coverage/gift-wrapping.md]: Change to declarative

File: design-documents/graph-ql/coverage/gift-wrapping.md [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+
+```graphql
+###### Begin: Extending existing types ######
+type CartItemUpdateInput {
+    gift_wrapping_id: ID @doc(description: "The unique identifier of the gift wrapping to be used for the cart item")
+    gift_message: GiftMessageInput @doc(description: "Gift message details for the cart item")
+}
+
+type Mutation {
+    setGiftOptionsOnCart(cart_id: String!, gift_message: GiftMessageInput, gift_wrapping_id: ID, include_gift_receipt: Boolean, include_printed_card: Boolean): SetGiftOptionsOnCartOutput @doc(description: "Set gift options like gift wrapping or gift message for the entire cart")
```
Comment (paliarush) [inline design-documents/graph-ql/coverage/gift-wrapping.md]: Change to declarative

File: design-documents/graph-ql/coverage/gift-wrapping.md (hunk 5 above)
Comment (paliarush) [reply design-documents/graph-ql/coverage/gift-wrapping.md:16]: Done

File: design-documents/graph-ql/coverage/gift-wrapping.md (hunk 7 above)
Comment (paliarush) [reply design-documents/graph-ql/coverage/gift-wrapping.md:50]: Done

File: design-documents/graph-ql/coverage/gift-wrapping.md (hunk 8 above)
Comment (paliarush) [reply design-documents/graph-ql/coverage/gift-wrapping.md:51]: Done

File: design-documents/graph-ql/coverage/gift-wrapping.md (hunk 9 above)
Comment (paliarush) [reply design-documents/graph-ql/coverage/gift-wrapping.md:181]: Done

**Pull request merged**